### PR TITLE
h2olog: decode sockaddr in USDT parameters

### PR DESCRIPTION
**Pull request by gfx**
Adding a feature to decode `struct sockaddr*` in h2olog.

The results is something like this:

```json
{"type":"h3-packet-receive","seq":1195,"dest-family":2,"dest-addr":"127.0.0.1","dest-port":35798,"src-family":2,"src-addr":"127.0.0.1","src-port":46087,"bytes-len":63,"time":1605163527393}
```

### CODE REVIEW

File: src/h2olog/h2olog.h [hunk 1]
```diff
@@ -29,8 +29,24 @@
...
 }
 
+// Corresponding to the types in the BPF program.
+// In particualr, h2olog_sockaddr_storage is limited only to
+// sockaddr_in and sockaddr_in6 because `struct sockaddr_storage`
+// is too large to be put on the BPF stack (limited to 512 bytes),
+// while `sizeof(struct sockaddr_storage)` is 128 bytes in Linux/x64.
+typedef sockaddr h2olog_sockaddr;
+typedef sockaddr_in h2olog_sockaddr_in;
+typedef sockaddr_in6 h2olog_sockaddr_in6;
```
Comment (kazuho) [inline src/h2olog/h2olog.h:43]: I do not think we need these lines, do we? This header file is for C++, and IIRC there's no need to have typedefs.

File: src/h2olog/h2olog.h [hunk 2]
```diff
@@ -29,8 +29,24 @@
...
+typedef union {
+    h2olog_sockaddr sa;
+    h2olog_sockaddr_in sin;
+    h2olog_sockaddr_in6 sin6;
+} h2olog_sockaddr_storage;
```
Comment (kazuho) [inline src/h2olog/h2olog.h:48]: Here also. `union h2o_log_sockaddr_storage { ... };` should be sufficient.

File: src/h2olog/json.cc [hunk 3]
```diff
@@ -1,6 +1,14 @@
+#include "h2olog.h"
 #include "json.h"
+
+extern "C" {
+#include <stdio.h>
```
Comment (kazuho) [inline src/h2olog/json.cc:5]: use cstdio?

**Review by kazuho — COMMENTED** [review]
Thank you for the PR. The PR looks mostly good to me.

Regarding the output format, would it make sense to change the encoding to something like `"source": "1.2.3.4:12345", "destination": "[2001:0db8:85a3::8a2e:0370:7334]:12345"`?

IMO most address decoders would be capable of decoding these notations using the implied address family.

**Comment by hfujita** [comment]
Wow this will be super useful. Thanks!

**Comment by gfx** [comment]
@hfujita Sorry to be late. It's quite complicated!

@kazuho Changed sockaddr stringify format!: https://github.com/h2o/h2o/pull/2486/commits/67c5d6e9ea165783f7da3980e448e0c48a3d93c6

File: src/h2olog/h2olog.h (hunk 1 above)
Comment (gfx) [reply src/h2olog/h2olog.h]: b557aaad8d0789e97f7ed05ce885c57bf7309d11

Indeed, It's no need to define the confusing type names. Now it becomes clear and tidy.

**Review by kazuho — APPROVED** [review]
Looking great. @gfx Thank you for your efforts.

**Comment by gfx** [comment]
@kazuho Thanks for reviewing! 😄

**Pull request merged**